Hospital/HospitalApi/Controllers/EventController.cs

```diff
@@ -30,14 +34,45 @@ public IActionResult AddEvent(EventDTO newEvent)
         {
             var incomingEvent = new StoredEvent()
             {
-                StateData = newEvent.StateData,
+                Step = newEvent.Step,
                 Time = DateTime.Now,
-                UserId = newEvent.UserId
+                Username = newEvent.Username
             };
 
             uow.GetRepository<IStoredEventWriteRepository>().Add(incomingEvent);
 
             return Ok();
         }
+
+        [Authorize(Roles = "Manager")]
```

Comment:
If all other endpoints should be authorized by the same role it is possible to put the authorize attribute on the whole controller (bellow Route attribute) and then for the one endpoint that doesnt require authorization we could put attribute [AllowAnonymous]